File: Python/programmers/etc/level2/타겟_넘버.py
def solution(numbers, target):

    def dfs(numbers, target, idx, cur):
        if idx == len(numbers):
            return 1 if cur == target else 0
        return dfs(numbers, target, idx+1, cur+numbers[idx]) + dfs(numbers, target, idx+1, cur-numbers[idx])
    
    return dfs(numbers, target, 0, 0)


numbers = [1, 1, 1, 1, 1]
target = 3
print(solution(numbers, target)) # 5

numbers = [4, 1, 2, 1]
target = 4
print(solution(numbers, target)) # 2

File: Java/programmers/etc/level2/타겟넘버.java
package CodeTest.Java.programmers.etc.level2;

public class 타겟넘버 {
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }

    private static int dfs(int[] numbers, int target, int idx, int cur) {
        if (idx == numbers.length) {
            return cur == target ? 1 : 0;
        }
        return dfs(numbers, target, idx+1, cur+numbers[idx]) + dfs(numbers, target, idx+1, cur-numbers[idx]);
    }
}
